src/components/date/date.test.js

```diff
@@ -0,0 +1,545 @@
+    CypressMountWithProviders(
+      <Confirm open height="60px">
+        <DateInputCustom disablePortal={state} />
+      </Confirm>
+    );
+
+    dateInput().click();
+    dayPickerWrapper().should(visibility);
+  });
+
+  describe("check events for TextEditor component", () => {
+    let callback;
+    const text = "1";
+
+    beforeEach(() => {
+      callback = cy.stub();
+    });
+
+    it("should call onChange callback when a clear event is triggered", () => {
+      CypressMountWithProviders(
+        <DateInputCustom text={text} onChange={callback} />
+      );
+
+      dateInput()
+        .clear()
+        .then(() => {
+          // eslint-disable-next-line no-unused-expressions
+          expect(callback).to.have.been.calledOnce;
+        });
+    });
+
+    it("should call onChange callback when a type event is triggered", () => {
+      CypressMountWithProviders(
+        <DateInputCustom text={text} onChange={callback} />
+      );
+
+      dateInput()
+        .type("1")
+        .then(() => {
+          // eslint-disable-next-line no-unused-expressions
+          expect(callback).to.have.been.calledOnce;
+        });
+    });
+
+    it.each([
+      ["en-US", "05/12/2022", enUS],
+      ["en-CA", "05/12/2022", enCA],
+      ["en-ZA", "12/05/2022", enZA],
+      ["de", "12.05.2022", de],
+      ["es-ES", "12/05/2022", es],
+      ["fr-FR", "12/05/2022", fr],
+      ["fr-CA", "12/05/2022", frCA],
+      ["zh-CN", "2022/05/12", zhCN],
+      ["pl-PL", "12.05.2022", pl],
+      ["bg-BG", "12.05.2022", bg],
+      ["zh-HK", "12/05/2022", zhHK],
+      ["hu-HU", "2022. 05. 12.", hu],
+      ["fi-FI", "12.05.2022", fi],
+      ["de-AT", "12.05.2022", deAT],
+      ["ko-KR", "2022. 05. 12.", ko],
+      ["ar-EG", "12/05/2022", arEG],
+      ["hi-HI", "12/05/2022", hi],
+      ["sl-SI", "12. 05. 2022", sl],
+      ["lv", "12.05.2022.", lv],
+    ])(
+      "should use %s locale and change the formattedValue to %s after selecting date",
+      (localeValue, formattedValueParam, dateFnsLocaleValue) => {
+        CypressMountWithProviders(
+          <DateInputCustom onChange={callback} />,
+          sageTheme,
+          {
+            locale: () => localeValue,
+            date: {
+              dateFnsLocale: () => dateFnsLocaleValue,
+            },
+          }
+        );
+        const getDatetoEnter = () => {
+          if (["en-US", "en-CA"].includes(localeValue))
+            return MMDDYYYY_DATE_TO_ENTER;
+          if (["zh-CN", "hu-HU", "ko-KR"].includes(localeValue))
+            return YYYYMMDD_DATE_TO_ENTER;
+
+          return DDMMYYY_DATE_TO_ENTER;
+        };
+
+        dateInput().clear().type(getDatetoEnter(), { delay: 0 });
+        dateInput().click();
+        dayPickerByText("12").click();
+
+        dateInput()
+          .wait(250)
+          .should("have.attr", "value", formattedValueParam)
+          .then(() => {
+            expect(callback.getCalls()[12].args[0].rawValue).to.equals(
```

Comment:
Instead of using a magic number to get the last call on the stack, we could use a util function like so

```suggestion
           const getLastCall = (calls) => calls.pop();
...
            expect(getLastCall(callback.getCalls()).args[0].rawValue).to.equals(
```

Reply:
it isn't a last call always :)

Reply:
They are as it's the last call you're testing, even `callback.getCalls()[0]` is the last call on the stack is you try it out

Reply:
Happy to approve this now and we can add the above as a util so it can be shared in other test files at a later date